File: answers/leetcode/Search a 2D Matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """

        row = 0
        column = len(matrix[row]) - 1
        while row >= 0 and row < len(matrix) and column >= 0 and column < len(matrix[row]):
            if matrix[row][column] == target:
                return True
            elif matrix[row][column] < target:
                row += 1
            else:
                column -= 1
        return False

File: answers/leetcode/Search a 2D Matrix.cpp
//@result 134 / 134 test cases passed. Status: Accepted Runtime: 12 ms Submitted: 0 minutes ago You are here! Your runtime beats 43.59% of cpp submissions.

class Solution {
public:
    bool searchMatrix(vector<vector<int> > &matrix, int target) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int rows = matrix.size();
        int cols = matrix[0].size();
        int low = 0;
        int high = rows * cols - 1;
        while (low <= high)
        {
			int mid = (low + high) / 2;
			if (matrix[mid / cols][mid % cols] == target)
			{
				return true;
			} else if (matrix[mid / cols][mid % cols] < target)
			{
				low = mid + 1;
			} else if (matrix[mid / cols][mid % cols] > target)
			{
				high = mid - 1;
			}
		}
		return false;
    }
};
